kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/RestResourceQuarkusTemplate.java

```diff
@@ -88,10 +91,23 @@ public class $Type$Resource {
     }
 
     @GET
-    @Produces(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, MediaType.TEXT_HTML})
     @Operation(operationId = "getAllProcessInstances_$name$", summary = "$documentation$", description = "$processInstanceDescription$")
-    public List<$Type$Output> getResources_$name$() {
-        return processService.getProcessInstanceOutput(process);
+    public Response getResources_$name$(@Context HttpHeaders headers) {
+        List<$Type$Output> out = processService.getProcessInstanceOutput(process);
+        boolean wantsHtml = headers.getAcceptableMediaTypes()
+            .stream()
+            .anyMatch(mt -> mt.isCompatible(MediaType.TEXT_HTML_TYPE)
+                    && !mt.isWildcardType() && !mt.isWildcardSubtype());
```

Comment:
MediaType.isCompatible(TEXT_HTML_TYPE) already excludes generic */*, so these extra checks are redundant.
```suggestion
                  );
```